app/interactors/users/generate_access_token.rb

```diff
@@ -0,0 +1,27 @@
+module Users
+  class GenerateAccessToken
+    include Interactor
+
+    delegate :user, to: :context
+
+    def call
+      context.access_token = access_token
+    end
+
+    private
+
+    def access_token
+      JWT.encode(payload, jwt_secret, "HS256")
+    end
+
+    def jwt_secret
+      ENV.fetch("JWT_SECRET")
+    end
+
+    def payload
+      {
+        sub: user.id
```

Comment:
It doesn't work if user `nil`

Reply:
It might be a solution?
```
    def access_token
      if interactor.context.user.present?
        JWT.encode(payload, jwt_secret, "HS256")
      else
        JWT = nil
      end
    end
```

Reply:
It might be a solution?
~~~
    def access_token
      if interactor.context.user.present?
        JWT.encode(payload, jwt_secret, "HS256")
      else 
        JWT.encode(nil)
      end
    end
~~~

Reply:
@w1nway